src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs

```diff
@@ -0,0 +1,169 @@
+            // we are building the regex that matches
+            // 1. it starts with a lower case letter or a digit which is considered as the end of its previous word
+            //    or it is at the beginning of this string
+            //    or we hit a word separator including \W, _, ., @, -, spaces
+            // 2. it either should match one of the candidates in the rename rules dictionary
+            // 3. it should followed by a upper case letter which is considered as the beginning of next word
+            //    or it is the end of this string
+            //    or we hit a word separator including \W, _, ., @, -, spaces
+            _regex = new Regex(@$"([\W|_|\.|@|-|\s|\$\da-z]|^)({regexRawString})([\W|_|\.|@|-|\s|\$A-Z]|$)");
+        }
+
+        public static void UpdateAcronyms(this IEnumerable<Schema> allSchemas, IReadOnlyDictionary<string, string> renameRules)
+        {
+            ApplyRenameRules(renameRules);
+            foreach (var schema in allSchemas)
+            {
+                TransformSchema(schema);
+            }
+        }
+
+        private static void TransformSchema(Schema schema)
+        {
+            switch (schema)
+            {
+                case ChoiceSchema:
+                case SealedChoiceSchema:
+                    TransformBasicSchema(schema);
+                    break;
+                case ObjectSchema objSchema: // GroupSchema inherits ObjectSchema, therefore we do not need to handle that
+                    TransformObjectSchema(objSchema);
+                    break;
+                default:
+                    throw new InvalidOperationException($"Unknown schema type {schema.GetType()}");
+            }
+        }
+
+        private static void TransformBasicSchema(Schema schema)
+        {
+            TransformLanguage(schema.Language);
+        }
+
+        private static void TransformLanguage(Languages languages)
+        {
+            var originalName = languages.Default.Name;
+            if (_cache.TryGetValue(originalName, out var result))
+            {
+                languages.Default.Name = result;
+                return;
+            }
+            result = originalName.EnsureNameCase();
+            languages.Default.Name = result;
+            _cache.TryAdd(originalName, result);
+        }
+
+        private static void TransformObjectSchema(ObjectSchema objSchema)
+        {
+            // transform the name of this schema
+            TransformBasicSchema(objSchema);
+            foreach (var property in objSchema.Properties)
+            {
+                TransformLanguage(property.Language);
+            }
+        }
+
+        private static IDictionary<string, string> RenameRules = new Dictionary<string, string>();
+
+        internal static string EnsureNameCase(this string name)
+        {
+            if (_regex == null)
+                return name; // this means we do not have any replacement rules
```

Comment:
It also maybe because method is called out of order. Please differentiate these two cases.

Reply:
I changed the signature by adding renameRules dictionary to the function signature, please take another look